confd/pkg/resource/template/template_funcs_test.go

```diff
@@ -59,8 +59,10 @@ func Test_BGPFilterBIRDFuncs(t *testing.T) {
 		ImportV4: []v3.BGPFilterRuleV4{
 			{Action: "Accept", Source: "RemotePeers", Interface: "vxlan.calico", MatchOperator: "NotIn", CIDR: "55.4.0.0/16"},
 			{Action: "Reject", Source: "RemotePeers", MatchOperator: "NotIn", CIDR: "55.4.0.0/16"},
+			{Action: "Reject", Source: "RemotePeers", MatchOperator: "NotIn", CIDR: "55.4.0.0/16", PrefixLength: &v3.BGPFilterPrefixLengthV4{Min: int32Helper(16), Max: int32Helper(24)}},
 			{Action: "Reject", Interface: "eth0", MatchOperator: "NotIn", CIDR: "55.4.0.0/16"},
 			{Action: "Accept", Interface: "eth0", Source: "RemotePeers"},
+			{Action: "Reject", Interface: "eth0", Source: "RemotePeers", PrefixLength: &v3.BGPFilterPrefixLengthV4{Min: int32Helper(16), Max: int32Helper(24)}},
```

Comment:
If I'm reading this correctly, including a PrefixLength has no impact on the resulting filter if there is no CIDR.

I think we should add a check to this function: https://github.com/projectcalico/calico/blob/master/libcalico-go/lib/validator/v3/validator.go#L1396-L1415

That returns an error if PrefixLength is specified by CIDR is not. Best not to allow configurations that don't have any effect in order to avoid user confusion.